system/doc/efficiency_guide/maps.xml

```diff
@@ -0,0 +1,618 @@
+  </list>
+
+  <section>
+    <title>Using Maps as an Alternative to Records</title>
+
+    <list type="bulleted">
+      <item><p>Use the map syntax instead of the functions in
+      the <seeerl marker="stdlib:maps">maps</seeerl> module.</p></item>
+
+      <item><p>Avoid having more than 32 elements in the
+      map. As soon as there are more than 32 elements in the map,
+      it will require more memory and keys can no longer be shared
+      with other instances of the map.</p></item>
+
+      <item><p>When creating a new map, always create it with the
+      all the keys that will ever be used. To maximize sharing of
+      keys (thus minimizing memory use), create a single function
+      that constructs the map using the map syntax and always use
+      it.</p></item>
+
+      <item><p>Always update the map using the
+      <c>:=</c> operator (that is, requiring that an element with
+      that key already exists).  The <c>:=</c> operator is slightly
+      more efficient, and it will help catch mispellings of
+      keys.</p></item>
+
+      <item><p>Whenever possible, match multiple map elements at
+      once.</p></item>
+
+      <item><p>Whenever possible, update multiple map elements at
+      once.</p></item>
+
+      <item><p>Avoid default values and the <seemfa
+      marker="stdlib:maps#get/3">maps:get/3</seemfa>
+      function.</p></item>
+
+      <item><p>To avoid having to deal with a map that may lack some keys,
+      <seemfa marker="stdlib:maps#merge/2">maps:merge/2</seemfa>
+      can efficiently add multiple default values. For example:</p>
+      <code type="erl"><![CDATA[
+      DefaultMap = #{shoe_size => 42, editor => emacs},
+      MapWithDefaultsApplied = maps:merge(DefaultMap, OtherMap)]]></code>
+      </item>
+    </list>
+  </section>
+
+  <section>
+    <title>Using Maps as Dictionaries</title>
+
+    <p>Using a map as a dictionary implies the following usage pattern:</p>
+    <list type="bulleted">
+      <item><p>Keys are usually variables not known at compile-time.</p></item>
+      <item><p>There can be any number of elements in the map.</p></item>
+      <item><p>Usually, no more than one element is looked up or
+      updated at the time.</p></item>
+    </list>
+
+    <p>Given that usage pattern, there is usually less performance to
+    be gained by using the map syntax compared to the <c>maps</c>
+    module. Therefore, which one to use is mostly a matter of
+    taste.</p>
+
+    <p>Maps are usually the most efficient dictionary data structure,
+    with a few possible exceptions:</p>
+
+    <list type="bulleted">
+      <item><p>If it is necessary to frequenquently convert a
+      dictionary to a sorted listed, or from a sorted list to a
+      dictionary, using <seeerl marker="stdlib:gb_trees">gb_trees</seeerl>
+      could be a better choice.</p></item>
+
+      <item><p>If all keys are non-negative integers, the <seeerl
+      marker="stdlib:array">array</seeerl> module could be a better
```

Comment:
I think the technical writers prefer "can" before "could", but I haven't bothered to check their document...

Reply:
Yes, that sounds vaguely familiar. I'll change.